Directory.Build.targets

```diff
@@ -402,6 +383,32 @@ if (version == null)
 UnityVersion = version.Substring("m_EditorVersion: ".Length);
 
 Log.LogMessage("Unity Version: " + version);
+]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+    <!-- Locate the TestRunner.dlls by filling the wildcard with the template version number. 3d is the default template -->
+  <UsingTask TaskName="LocateTestRunner" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <UnityLibcache ParameterType="System.String" Required="true" />
+      <TestRunnerPath ParameterType="System.String" Output="true" />
+    </ParameterGroup>
+
+    <Task>
+      <Using Namespace="System.IO" />
+      <Using Namespace="System.Linq" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+var directories = Directory.GetDirectories(UnityLibcache, "com.unity.template.3d-*");
+if (directories.Length == 1)
+{
+    TestRunnerPath = Path.Combine(directories[0], "ScriptAssemblies");
+    Log.LogMessage("Found TestRunner path at: " + TestRunnerPath);
+    return true;
+}
+
+Log.LogError("Failed to resolve 'com.unity.template.3d-*' for TestRunner at: " + UnityLibcache);
```

Comment:
Is the package name hard-coded here a possible pitfall?

Reply:
fine by me, if it breaks in a future Unity version, it's an easy fix.